### PR TITLE
Remove unnecessary boost system dependency

### PR DESCRIPTION
**Pull request by nachovizzo**
This library dependency was not being used anywhere, and therefore
adding an additional dependency to the whole build system for no reason.

Signed-off-by: Ignacio Vizzo <ignaciovizzo@gmail.com>

### CODE REVIEW

**Review by Idclip — APPROVED** [review]
Thanks @nachovizzo, good spot! Looks good to me. Note that I think iostreams still depends on system (https://pdimov.github.io/boostdep-report/boost-1.66.0/iostreams.html) but we don't list these secondary dependencies anyway

**Comment by nachovizzo** [comment]
> Thanks @nachovizzo, good spot! Looks good to me. Note that I think iostreams still depends on system (https://pdimov.github.io/boostdep-report/boost-1.66.0/iostreams.html) but we don't list these secondary dependencies anyway

Agree.

By the way... I'd love to get rid of the `boost::iostreams` dependency for real. Since is the only library I need to install in order to build my application. And depending on the target system(CentOS, Debian, etc).... this varies a lot. Getting rid of the iostreams dependency would simplify my life quite a lot...

Is there any interest in addressing this?

**Comment by Idclip** [comment]
> By the way... I'd love to get rid of the boost::iostreams dependency for real. Since is the only library I need to install in order to build my application. And depending on the target system(CentOS, Debian, etc).... this varies a lot. Getting rid of the iostreams dependency would simplify my life quite a lot...

We're totally on board with this and it's something we've been slowly working towards. The main complexity involved in this is with boost::interprocess. Other components such as uuid and filesystem aren't as tricky, but until we have a replacement for interprocess there's not much point acting on the other components. We also have other work which adds other boost dependencies (see the feature/sharpening branch) and of course, boost::python. 

In summary, absolutely, and any work contributed to help remove boost would be welcomed!

**Review by jmlait — APPROVED** [review]
